src/lib/storage/run_length_segment/run_length_segment_iterable.hpp

```diff
@@ -52,13 +52,13 @@ class RunLengthSegmentIterable : public PointAccessibleSegmentIterable<RunLength
     using EndPositionIterator = typename pmr_vector<ChunkOffset>::const_iterator;
 
    public:
-    explicit Iterator(const ValueIterator& value_it, const NullValueIterator& null_value_it,
-                      const EndPositionIterator& end_position_it, const EndPositionIterator& end_position_begin_it,
+    explicit Iterator(const ValueIterator&& value_it, const NullValueIterator&& null_value_it,
```

Comment:
Not sure what you are trying to achieve here. As `Iterator` is always constructed with a prvalue, it should take its argument by value and move them into the members.

Reply:
I personally don't like constructors that potentially create copies or corrupt the caller's data. From my (limited) understanding, we simply better enforce that prvalues are passed and that we cannot run into certain situations.

Reply:
I'd only address this in cases where it actually becomes an issue. 90%+ of developers don't get universal references right (and I don't necessarily count myself towards the other 10%).

Reply:
Your call then. You have a much greater experience with students working on Hyrise.
If you think it might confuse people, I will only keep the changes that we always move when possible and through out the `&&`.

Reply:
I'd take it by value.